src/index.ts

```diff
@@ -15,10 +15,24 @@ import {
 import { Code, code, imp, Import } from "ts-poet";
 import PluginOutput = Types.PluginOutput;
 
+const DEFAULT_IMPORT_NAME = "GraphQLAutoGeneratedTypes";
+
 /** Generates `newProject({ ... })` factory functions in our `graphql-types` codegen output. */
 export const plugin: PluginFunction = async (schema, documents, config: Config) => {
   const chunks: Code[] = [];
 
+  if (!!config.typeDeclarationFilePath) {
+    let typeImportPath = config.typeDeclarationFilePath;
+    if (typeImportPath.endsWith(".ts") || typeImportPath.endsWith(".js")) {
+      typeImportPath = typeImportPath.slice(0, -3);
+    } else if (typeImportPath.endsWith(".tsx") || typeImportPath.endsWith(".jsx")) {
+      typeImportPath = typeImportPath.slice(0, -4);
+    }
```

Comment:
Hm, one of the up-coming wrinkles we have is that true ESM modules actually want to keep file names in imports (which is weird, but :shrug: ). It makes me wonder if maybe instead of the file path, we should have the property just name `typeImportPath` and then the user would set it to `typeImportPath: ./types` if they want the extension-less import, or they really could do `typeImportPath: ./types.ts` if they wanted to stay ESM compliant.

Granted, I might be looking too far ahead, but wdyt?